### PR TITLE
Implement internal cluster

### PR DESCRIPTION
**Pull request by vaijab**
See https://github.com/UKHomeOffice/keto/issues/71

Would be awesome if people could test this works as expected.

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -186,20 +196,22 @@ func (c *Controller) clusterExists(name string, cl cloudprovider.Clusters) (bool
...
-	}
 	pooler, impl := c.Cloud.NodePooler()
 	if !impl {
 		return ErrNotImplemented
 	}
 
-	// TODO Check if a masterpool for given cluster exists first? Maybe we
-	// don't care if masterpool does not exist yet, as long as the cluster infra exists?
-	if exists, err := c.clusterExists(p.ClusterName, cl); !exists {
+	clusters, err := c.GetClusters(p.ClusterName)
```
Comment (lewismarshall) [inline pkg/controller/controller.go]: Maybe a single method c.GetCluster wrapping GetClusters or rename GetClusters (as it's redundant as there's no wildcard option)...

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -186,20 +196,22 @@ func (c *Controller) clusterExists(name string, cl cloudprovider.Clusters) (bool
...
+	if err != nil {
 		return err
 	}
+	if len(clusters) != 1 {
+		return ErrClusterDoesNotExist
+	}
+	if len(clusters) > 1 {
+		return fmt.Errorf("more than one cluster found matching %q name", p.ClusterName)
```
Comment (lewismarshall) [inline pkg/controller/controller.go]: Seems like a single place to raise this error (see comment above)...

Comment (vaijab) [inline pkg/controller/controller.go]: When `GetClusters` returns no results, this does not mean there is a problem. It's up to a caller to interpret that, hence the error above.

**Comment by KashifSaadat** [comment]
**Internal Flag Testing:**
- [ ] Test create internal cluster with master & compute nodes (Successful)
- [ ] Test create external cluster with master & compute nodes, validate no regression (Successful)
- [ ] Test create internal cluster with 0 compute nodes, separately add compute pool and confirm it is internal
- [ ] Test create internal cluster with compute pool on a VPC with NAT Gateway, validate external connectivity

File: pkg/controller/controller.go (hunk 1 above)
Comment (lewismarshall) [inline pkg/controller/controller.go]: Fair enough. TODO: Add a single GetCluster call that will error if name doesn't match.

**Pull request merged**